bin/millau/runtime/src/lib.rs

```diff
@@ -309,6 +309,22 @@ impl pallet_substrate_bridge::Config for Runtime {
 	type BridgedChain = bp_rialto::Rialto;
 }
 
+parameter_types! {
+	// One hundred and fifty old headers is ~15 mins (assuming a target block time of 6s). This
+	// should give the relayer enough time to get a justification for a header.
+	//
+	// Since the ancestry checker we're using doesn't need to access to storage for verification
+	// this can be fairly high.
+	pub const MaxHeadersInSingleProof: u8 = 150;
```

Comment:
same as below - instead of setting this arbitrarily, we should make it depend on the session length. For small header sizes (or larger block times/weights) it might be totally possible to deliver a proof that spans the entire session in a single transaction.